drivers/stepper/Kconfig

```diff
@@ -18,35 +18,18 @@ config STEPPER_INIT_PRIORITY
 	help
 	  Stepper motor controller initialization priority.
 
+config STEPPER_ASYNC_API
+	bool "Asynchronous Stepper API"
+	default y
+	help
+	  Enable asynchronous stepper API.
+
```

Comment:
Is a config necessary? Even with the config, a user can not provide a callback.

Reply:
Yep this config is necessary. For instance, the periodic polling as done for now in trinamic driver depends on this config.

IMO in your application if you need Async event feedback, it should be possible to switch it on through one single config and drivers can decide if they want to support it or not.

Prolly one could later use this config in app code to toggle event processing.

Reply:
Not sure if a config per say is necessary there. There are a lot of callbacks presently which don't need a specific config, for example setting can filters, or [rtc callbacks](https://docs.zephyrproject.org/apidoc/latest/group__rtc__interface.html). Its left to the end user to decide, if they want to register a callback or not.

Also, in the case of the trinamic driver, or the gpio driver, if not callback is registered, you can avoid submitting work, since there's no work to be done.